src/core/radio/ble_secure.cpp

```diff
@@ -0,0 +1,542 @@
+/*
+ *  Copyright (c) 2023, The OpenThread Authors.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions are met:
+ *  1. Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *  2. Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *  3. Neither the name of the copyright holder nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "ble_secure.hpp"
+
+#if OPENTHREAD_CONFIG_BLE_TCAT_ENABLE
+
+#include <openthread/platform/ble.h>
+#include "common/instance.hpp"
+#include "common/locator_getters.hpp"
+#include "common/log.hpp"
+#include "common/tlvs.hpp"
+#include "meshcop/secure_transport.hpp"
+
+using namespace ot;
+
+/**
+ * @file
+ *   This file implements the secure Ble agent.
+ */
+
+namespace ot {
+namespace Ble {
+
+RegisterLogModule("BleSecure");
+
+BleSecure::BleSecure(Instance &aInstance)
+    : InstanceLocator(aInstance)
+    , mTls(aInstance, false, false)
+    , mTcatAgent(aInstance)
+    , mTlvMode(false)
+    , mReceivedMessage(nullptr)
+    , mSendMessage(nullptr)
+    , mTransmitTask(aInstance)
+    , mBleState(kStopped)
+    , mMtuSize(kInitialMtuSize)
+{
+}
+
+Error BleSecure::Start(ConnectCallback aConnectHandler, ReceiveCallback aReceiveHandler, bool aTlvMode, void *aContext)
+{
+    Error error = kErrorNone;
+
+    VerifyOrExit(mBleState == kStopped, error = kErrorAlready);
+
+    mConnectCallback.Set(aConnectHandler, aContext);
+    mReceiveCallback.Set(aReceiveHandler, aContext);
+    mTlvMode = aTlvMode;
+    mMtuSize = kInitialMtuSize;
+
+    SuccessOrExit(error = otPlatBleEnable(&GetInstance()));
+    SuccessOrExit(error = otPlatBleGapAdvStart(&GetInstance(), 0));
```

Comment:
Here the parameter 0 is not within `OT_BLE_ADV_INTERVAL_MIN` and `OT_BLE_ADV_INTERVAL_MAX`, so that the platform gives an Error 7 InvalidArgs when typing `tcat start`. Maybe `0` was intended as selecting a default value chosen by the platform ?  One idea is that `tcat start` could have an int parameter with this value. If omitted, `0` is passed to this function, or otherwise an appropriate default adv-interval.

Reply:
fixed.